pkg/teller_test.go

```diff
@@ -68,17 +69,51 @@ func (im *InMemProvider) ProviderHumanToMachine() map[string]string {
 func (im *InMemProvider) Name() string {
 	return "inmem"
 }
-func NewInMemProvider(alwaysError bool) (Providers, error) {
+
+func (im *InMemProvider) Meta() core.MetaInfo {
+	return core.MetaInfo{}
+}
+
+//nolint
+func init() {
+	inmemProviderMeta := core.MetaInfo{
+		Name:        "inmem-provider",
+		Description: "test-provider",
+	}
+
+	inmemProviderErrorMeta := core.MetaInfo{
+		Name:        "inmem-provider-error",
+		Description: "test-provider-error",
+	}
+
+	providers.RegisterProvider(inmemProviderMeta, NewInMemProvider)
+	providers.RegisterProvider(inmemProviderErrorMeta, NewInMemErrorProvider)
```

Comment:
we used inmem provider only in test, so it's builder wasn't following the builder standard, I changed it so the test will register 2 providers for inmem tests:
1. When we want it to work normally - alwaysError: false
2. When we want it to have error - alwaysError: true
When resolving the first one - we are just using the `inmem-provider` key
When we want to test with errors we resolve the other - `inmem-provider-error` key